ui/raidboss/raidemulator.js

```diff
@@ -0,0 +1,832 @@
+        return 1;
+      return 0;
+    });
+
+    // Clear party
+    while (this.partyElement.firstChild)
+      this.partyElement.firstChild.remove();
+
+    for (let i = 0; i < party.length; i++) {
+      let playerElement = document.createElement('div');
+      playerElement.className = 'jobicon ' + party[i].job;
+      playerElement.id = party[i].id;
+      playerElement.setAttribute('name', party[i].name);
+      playerElement.setAttribute('job', party[i].job);
+      playerElement.addEventListener('click', function(e) {
+        let id = e.target.id;
+        let name = e.target.getAttribute('name');
+        let job = e.target.getAttribute('job');
+        gEmulatorView.logPlayer.SendPlayerEvent(name, job, id);
+      });
+      this.partyElement.appendChild(playerElement);
+    }
+  }
+
+  Analyze() {
+    // Clear all triggerlines
+    let triggerlines = document.getElementsByClassName('triggerline');
+    for (let i = 0; i < triggerlines.length; i++) {
+      while (triggerlines[i].firstChild)
+        triggerlines[i].firstChild.remove();
+    }
+
+    this.AnalyzeFight('all');
+    this.AnalyzeFight('tank');
+    this.AnalyzeFight('healer');
+    this.AnalyzeFight('dps');
+  }
+
+  AnalyzeFight(role) {
+    let triggers = gPopupText.triggers;
+    let data = {};
+    let logs = this.selectedFight.logs;
+
+    data.ShortName = function(name) {
+      return name.split(/\s/)[0];
+    };
+    // Reset local Suppression
+    this.suppressedTriggersLocal = {};
+
+    for (let i = 0; i < logs.length; i++) {
+      let log = logs[i];
+      for (let t = 0; t < triggers.length; t++) {
+        let trigger = triggers[t];
+        if (trigger.disabled)
+          continue;
+
+        let matches = log.match(trigger.localRegex);
+        if (matches != null)
+          this.RunTriggers(logs, log, trigger, role, data, matches);
+      }
+    }
+  }
+
+  RunTriggers(logs, log, trigger, role, data, matches) {
+    let run;
+    let result;
+    let timestamp = dateFromLogLine(log).getTime();
+    let delay = 0;
+    let suppress = 1;
+    let suppressed = this.suppressedTriggersLocal;
+
+    if (role == 'all')
+      suppressed = this.suppressedTriggers;
+
+    if (typeof trigger.delaySeconds === 'function')
+      delay = trigger.delaySeconds(data, matches) * 1000 || delay * 1000;
+    if (typeof trigger.delaySeconds === 'number')
+      delay = trigger.delaySeconds * 1000 || delay * 1000;
+
+    let startTime = dateFromLogLine(logs[0]).getTime();
+    let endTime = dateFromLogLine(logs[logs.length - 1]).getTime();
+    let pos = (timestamp + delay - startTime) / (endTime - startTime) * 100;
+    let players = {};
+    for (let p in this.players) {
+      if (this.players[p].role == role) {
+        Object.assign(players, {
+          [p]: {
+            name: this.players[p].name,
+            job: this.players[p].job,
+            role: this.players[p].role,
+          },
+        });
+      }
+    }
+    if (role == 'all')

```

Comment:
This is just a bug report from me, but looking at an o9s log from this last week, I found that it was ignoring all the "stack donut" and "get behind" all triggers in the analyzer.  I can send you the log file if that would help.

Reply:
Argh... no idea why it keeps doing that sometimes... does the fight start with water phase?
It may be some issue with the analyzer being faster than the raidboss module does load the triggers.
But honestly? No idea. This bug is driving me nuts.